<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1186</width>
    <height>929</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QWidget" name="widget_6" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>80</height>
           </size>
          </property>
          <widget class="QWidget" name="layoutWidget">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>1161</width>
             <height>51</height>
            </rect>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Load image</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="maximumSize">
               <size>
                <width>150</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Function</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="plane_1_label">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Plane 1</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="plane_2_label">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Plane 2</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="plane_3_label">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Plane 3</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="mixed_plane_label">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Mixed plane</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="MplCanvas" name="raw" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>200</width>
            <height>250</height>
           </size>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_4" native="true">
          <property name="maximumSize">
           <size>
            <width>149</width>
            <height>250</height>
           </size>
          </property>
          <widget class="QWidget" name="layoutWidget">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>141</width>
             <height>221</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QRadioButton" name="rgbBtn">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>RGB</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">plane_btns</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="cmyBtn">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>CMY</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">plane_btns</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="hsiBtn">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>HSI</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">plane_btns</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="xyzBtn">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>XYZ</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">plane_btns</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="labBtn">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>L*a*b*</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">plane_btns</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="yuvBtn">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>YUV</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">plane_btns</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="apply_1">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>14</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Apply</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="plane1" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>196</width>
            <height>250</height>
           </size>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="plane2" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>196</width>
            <height>250</height>
           </size>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="plane3" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>196</width>
            <height>250</height>
           </size>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="mixed_plane" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>196</width>
            <height>250</height>
           </size>
          </property>
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="MplCanvas" name="pseudo_processed" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_7" native="true">
          <widget class="QWidget" name="layoutWidget_3">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>0</y>
             <width>143</width>
             <height>211</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QRadioButton" name="autumn">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Autumn</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">pseudo_group</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="jet">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Jet</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">pseudo_group</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rainbow">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Rainbow</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">pseudo_group</string>
              </attribute>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QRadioButton" name="radioButton">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>100</width>
                  <height>28</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <family>Arial</family>
                  <pointsize>12</pointsize>
                 </font>
                </property>
                <property name="text">
                 <string>Custom</string>
                </property>
                <attribute name="buttonGroup">
                 <string notr="true">pseudo_group</string>
                </attribute>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="color_widget_1" native="true">
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>28</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">background-color: rgb(255, 255, 255);</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="color_widget_2" native="true">
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>28</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true">background-color: rgb(0, 0, 0);</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QSpinBox" name="c_level">
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>28</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="minimum">
               <number>2</number>
              </property>
              <property name="maximum">
               <number>5</number>
              </property>
              <property name="value">
               <number>2</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="apply_2">
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>28</height>
               </size>
              </property>
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>16</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Apply</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="pseudo1" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="pseudo2" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="pseudo3" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="mixed_pseudo" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="MplCanvas" name="kmeans_processed" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_3" native="true">
          <widget class="QWidget" name="layoutWidget_2">
           <property name="geometry">
            <rect>
             <x>30</x>
             <y>0</y>
             <width>141</width>
             <height>231</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QRadioButton" name="rgbBtn_3">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>RGB</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">btns_2</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="cmyBtn_3">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>CMY</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">btns_2</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="hsiBtn_3">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>HSI</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">btns_2</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="xyzBtn_3">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>XYZ</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">btns_2</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="labBtn_3">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>L*a*b*</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">btns_2</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="yuvBtn_3">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>YUV</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">btns_2</string>
              </attribute>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_7">
              <item>
               <widget class="QLabel" name="label_8">
                <property name="maximumSize">
                 <size>
                  <width>22</width>
                  <height>22</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <family>Arial</family>
                  <pointsize>12</pointsize>
                 </font>
                </property>
                <property name="text">
                 <string>k</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="k_level">
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>22</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <family>Arial</family>
                  <pointsize>12</pointsize>
                 </font>
                </property>
                <property name="minimum">
                 <number>2</number>
                </property>
                <property name="maximum">
                 <number>5</number>
                </property>
                <property name="value">
                 <number>2</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPushButton" name="apply_3">
              <property name="font">
               <font>
                <family>Arial</family>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Apply</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="kplane1" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="kplane2" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="kplane3" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="MplCanvas" name="mixed_kplane" native="true">
          <property name="maximumSize">
           <size>
            <width>210</width>
            <height>250</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MplCanvas</class>
   <extends>QWidget</extends>
   <header>mplcanvas.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="plane_btns"/>
  <buttongroup name="btns_2"/>
  <buttongroup name="pseudo_group"/>
  <buttongroup name="color"/>
 </buttongroups>
</ui>
